### DISCUSSION TITLE
Iced in a Virtual Machine

### DISCUSSION THREAD

**Discussion by chanchan69** [hecrj/iced#1321] [Q&A] [answered]
Iced does not render correctly in a virtual machine (VM) unless 3D acceleration is enabled ([for example in VirtualBox](https://media.discordapp.net/attachments/962090348090384464/967581176460083270/unknown.png)). For reference I am compiling Iced 0.3.0 using WGPU and am targeting 64 bit Windows clients. When I say that Iced is not working correctly I mean that it renders like [this](https://media.discordapp.net/attachments/962090348090384464/967575327603822743/unknown.png?width=1203&height=904) instead of like [this](https://media.discordapp.net/attachments/962805699304370226/968960285773811712/unknown.png?width=1609&height=904)

Enabling 3D acceleration for your VM is an easy solution if you are hosting the VM yourself (and your software supports it), however VMs that are hosted by a third party (think AWS, Google, Azure, etc) most likely cannot have 3D acceleration enabled as the user cannot access the VM client software. 

Is there a way to get Iced working correctly without 3D acceleration? Any suggestions are appreciated

**Comment by genusistimelord** [comment, marked as answer]
sadly no since it uses Vulkan/DX currently. Most of the GUI use WGPU in rust so this becomes rather hard to support none hardware accelerated VM's.  However if you use Web asm then it doesn't matter what the VM is as the end user accessing it via a web browser is what matters then.
